GitCommands/Git/GitModule.cs

```diff
@@ -3378,7 +3378,7 @@ internal GitBlame ParseGitBlame(string output, Encoding encoding)
             // is a blank line, and third is an introductory paragraph about the project.
 
             Dictionary<ObjectId, GitBlameCommit> commitByObjectId = new();
-            List<GitBlameLine> lines = new(capacity: 256);
+            List<GitBlameLine> lines = new(capacity: Math.Min(Math.Max(256, output.Length / 120), 1000));
```

Comment:
What are these magic numbers?

Reply:
It's a heuristic to size the list (to prevent multiple reallocation) depending on the size of the file content to blame with some limits to not allocate a too small list (256 as before) or too big (1000) one.
To 'estimate' the number of lines, I took a number bigger than what nearly all projects will have as mean source code length to guarantee that we will not over-allocate (because we blame nearly 90% of the time source code, right!?!) which is a length of 120 characters...

Reply:
It would be good to add some hints about this.
I remember looking at this line some minutes when I tried this commit out to understand it.
(Not affecting approval.)

Reply:
I have added a comment...